src/js_realm.hpp

```diff
@@ -213,6 +213,68 @@ class RealmDelegate : public BindingContext {
     friend class RealmClass<T>;
 };
 
+
+template<typename T>
+class ShouldCompactOnLaunchFunctor {
+public:
+    ShouldCompactOnLaunchFunctor(typename T::Context ctx, typename T::Function should_compact_on_launch_func)
+    : m_ctx(Context<T>::get_global_context(ctx))
+    , m_func(ctx, should_compact_on_launch_func)
+    , m_event_loop_dispatcher {ShouldCompactOnLaunchFunctor<T>::main_loop_handler}
+    , m_mutex{new std::mutex}
+    , m_cond_var{new std::condition_variable}
+    {
+    }
+
+    bool operator ()(const uint64_t total_bytes, const uint64_t used_bytes) {
```

Comment:
The function which is dispatched to the event loop in `EventLoopDispatcher` cannot return a value (the signature is `std::function<void(Args...)>`), and we use a shared variable between the threads.

Reply:
`main_loop_handler` can set a variable with the result and unlock the mutex. There is not need of two variables since there are only two threads involved and one of them is main which will be waiting for the result. this means there will be no chance of spurious wakeup. but you  can decide to keep the `m_should_compact_on_launch_callback_done` if you preferer.

so the pseudo code should be 
main thread
```
m_event_loop_dispatcher()
{
    std::unique_lock<std::mutex> lock(*m_mutex);
    m_cond_var->wait()
    return m_should_compact_on_launch;
}
```

and the loop_handler should be
```
Function::call
m_should_compact_on_launch = Value<T>::validated_to_boolean(ctx, ret_val);
this_object->m_should_compact_on_launch_callback_done = true;
this_object->m_cond_var->notify_one();
```

Reply:
this was handled